values.yaml

```diff
@@ -30,12 +31,21 @@ client:
   # It isn't common to need more than 2 client nodes.
   replicas: 2
   antiAffinity: "soft"
+
+  # The amount of RAM allocated to the JVM heap. This should be set to the
+  # same value as resources.requests.memory, or you may see OutOfMemoryErrors
+  # on startup.
+  heapMemory: 256m
```

Comment:
I'm a little bit confused: why is there a separate variable if it should be set to the same value as `resources.requests.memory`? Does it make sense to set it to a lower value with a very high resource limit/request?

Also, if it should fall back to `resources.requests.memory` I would probably not give it a default value, but use it in templates like this: `{{ default .Values.client.resources.requests.memory .Values.client.heapMemory }}`

This way `heapMemory` has no default value in values.yaml (less duplication and redundancy, explicit dependency between variables), but still possible to override. Of course, we have to document it in the readme, but I don't think it would be a commonly used feature anyway.

Reply:
Sorry, just realized the difference in units. Would have been nice though. 😄

Reply:
Lol yeah, my original thought was much like yours: to keep them identical. 

But that difference is units is a little annoying...